src/server/qgsservercachemanager.h

```diff
@@ -0,0 +1,172 @@
+/***************************************************************************
+                        qgsservercachemanager.h
+                        -----------------------
+
+  begin                : 2018-07-05
+  copyright            : (C) 2018 by René-Luc D'Hont
+  email                : rldhont at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSERVERCACHEMANAGER_H
+#define QGSSERVERCACHEMANAGER_H
+
+#include "qgsservercachefilter.h"
+#include "qgsaccesscontrol.h"
+#include "qgsserverrequest.h"
+
+#include <QMultiMap>
+#include <QDomDocument>
+#include "qgsproject.h"
+#include "qgis_server.h"
+#include "qgis_sip.h"
+
+SIP_IF_MODULE( HAVE_SERVER_PYTHON_PLUGINS )
+
+class QgsServerCachePlugin;
+
+
+/**
+ * \ingroup server
+ * \class QgsServerCacheManager
+ * \brief A helper class that centralizes caches accesses given by all the server cache filter plugins.
+ * \since QGIS 3.4
+ */
+class SERVER_EXPORT QgsServerCacheManager
+{
+#ifdef SIP_RUN
+#include "qgsservercachefilter.h"
+#endif
+
+  public:
+    //! Constructor
+    QgsServerCacheManager()
+    {
+      mPluginsServerCaches.reset( new QgsServerCacheFilterMap() );
+    }
```

Comment:
Considering that `qgsservercachemanager.cpp` exists, what do you think about moving the content of the constructor in this source file?

Same question about the content of the copy constructor, the destructor and the assignment operator.

Reply:
It's a copy of QgsAccessControl source files.